actionview/lib/action_view/template.rb

```diff
@@ -122,37 +122,42 @@ def self.finalize_compiled_template_methods=(_)
 
     extend Template::Handlers
 
-    attr_accessor :locals, :virtual_path
-
     attr_reader :source, :identifier, :handler, :original_encoding, :updated_at
-    attr_reader :variable, :format, :variant
+    attr_reader :variable, :format, :variant, :locals, :virtual_path
 
-    def initialize(source, identifier, handler, format: nil, variant: nil, **details)
+    def initialize(source, identifier, handler, format: nil, variant: nil, locals: nil, virtual_path: nil, updated_at: Time.now)
       unless format
         ActiveSupport::Deprecation.warn "ActionView::Template#initialize requires a format parameter"
         format = :html
       end
 
+      unless locals
+        ActiveSupport::Deprecation.warn "ActionView::Template#initialize requires a locals parameter"
+        locals = []
+      end
+
       @source            = source
       @identifier        = identifier
       @handler           = handler
       @compiled          = false
-      @original_encoding = nil
-      @locals            = details[:locals] || []
-      @virtual_path      = details[:virtual_path]
+      @locals            = locals
+      @virtual_path      = virtual_path
```

Comment:
How is the virtual_path built here, like it was with Path.build in decorate? I don’t see how decorate can be removed yet.

Reply:
1. [`find_templates` uses `Path.build` and passes that to `query`](https://github.com/rails/rails/blob/0e13941dbe9c4840175e5c6f5b2c3f572249a4ec/actionview/lib/action_view/template/resolver.rb#L210-L213)
2. [`query` passes the virtual path to the template constructor](https://github.com/rails/rails/blob/0e13941dbe9c4840175e5c6f5b2c3f572249a4ec/actionview/lib/action_view/template/resolver.rb#L222-L228)

Reply:
Ah, thanks! I was looking for the build_path method (which uses Path.build) and found no occurrences, so I got confused. Seems like we could kill that build_path private method. I’ll take a look later unless you want to look into it.

Reply:
You're totally right.  I'll rm that method.  Thank you!